src/main/java/com/gw2auth/oauth2/server/service/gw2account/Gw2AccountWithApiToken.java

```diff
@@ -0,0 +1,57 @@
+package com.gw2auth.oauth2.server.service.gw2account;
+
+import com.gw2auth.oauth2.server.repository.gw2account.Gw2AccountWithApiTokenEntity;
+import com.gw2auth.oauth2.server.service.Gw2ApiPermission;
+
+import java.time.Instant;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+
+public class Gw2AccountWithApiToken extends Gw2Account {
```

Comment:
Intuitively, I'd prefer composition here. This way, `Gw2Account` could also remain a record and both entities would _feel_ cleaner, imo.